src/main/java/greencity/service/impl/PlaceServiceImpl.java

```diff
@@ -88,13 +84,34 @@ public Place save(PlaceAddDto dto, String email) {
      * @author Kateryna Horokh
      */
     private void setPlaceToOpeningHours(Place place) {
-        log.info("in setPlaceToOpeningHours(Place place) - {}", place.getName());
+        log.info(LogMessage.SET_PLACE_TO_OPENING_HOURS, place.getName());
+
         List<OpeningHours> hours = place.getOpeningHoursList();
-        hours.forEach(
-            h -> {
-                h.setPlace(place);
-                openingHoursService.save(h);
-            });
+        checkRepeatingValue(hours);
+        hours.stream()
+            .distinct()
+            .forEach(
+                h -> {
+                    h.setPlace(place);
+                    openingHoursService.save(h);
+                });
+    }
+
+    /**
+     * Method for checking list of giving {@code OpeningHours} on repeating value of week days.
+     *
+     * @param hours - list of {@link OpeningHours} entity.
+     * @author Kateryna Horokh
+     */
+    private void checkRepeatingValue(List<OpeningHours> hours) {
+        log.info(LogMessage.CHECK_REPEATING_VALUE);
+        for (int i = 0; i < hours.size(); i++) {
+            for (int j = i + 1; j < hours.size(); j++) {
+                if (hours.get(i).getWeekDay() == (hours.get(j).getWeekDay())) {
```

Comment:
just a question to think of for you - not for changes here..
Why using == and not equals, and whats the difference if we're talking about enums here

again, just to think about